src/main/java/org/broadinstitute/consent/http/db/mapper/DataUseParser.java

```diff
@@ -0,0 +1,27 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import com.google.gson.Gson;
+import java.util.HashMap;
+import java.util.Map;
+import org.broadinstitute.consent.http.models.DataUse;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.broadinstitute.consent.http.util.gson.GsonUtil;
+
+public class DataUseParser implements ConsentLogger {
+
+  private final Gson gson = GsonUtil.gsonBuilderWithAdapters().create();
+  private final Map<String, DataUse> dataUseCache = new HashMap<>();
```

Comment:
I assume that the lifetime of a jdbi row reducer is only during a single request. If so, then this only improves the cases where the same data use expression occurs in different rows in the result, correct? Vs making the cache global across all operations. That would require using a concurrent map though so maybe it wouldn't be much of an improvement.

Reply:
Yes, mappers/reducers are instantiated per query, so their lifetime is short as is the cache - which I'm fine with for now. We certainly could make this a larger cache refactor but as you mention, we'd need to measure the tradeoffs. What we find in practice is that a lot of datasets have the exact same data use so I think the bulk of the performance gain comes from that fact.